### PR TITLE
Cleanup sparcv8leon3 port

### PR DESCRIPTION
**Pull request by lukileczo**
JIRA: RTOS-885

## Motivation and Context

Cleanup before GR740 port
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: hal/sparcv8leon3/gaisler/timer.c [hunk 1]
```diff
@@ -24,14 +24,20 @@
...
-
+/* Timer registers */
 
-/* clang-format off */
+#define GPT_SCALER     0       /* Scaler value register                 : 0x00 */
+#define GPT_SRELOAD    1       /* Scaler reload value register          : 0x04 */
+#define GPT_CONFIG     2       /* Configuration register                : 0x08 */
+#define GPT_LATCHCFG   3       /* Latch configuration register          : 0x0C */
+#define GPT_TCNTVAL(n) (n * 4) /* Timer n counter value reg (n=1,2,...) : 0xn0 */
+#define GPT_TRLDVAL(n) (n * 4) /* Timer n reload value register         : 0xn4 */
```
Comment (agkaminski) [inline hal/sparcv8leon3/gaisler/timer.c:34]: Shouldn't be `((n * 4) + 1)`?

Comment (lukileczo) [reply hal/sparcv8leon3/gaisler/timer.c:34]: In the documentation timers are numbered from 1, so as bad as it is I think we should stay true to it.

Comment (agkaminski) [reply hal/sparcv8leon3/gaisler/timer.c:34]: Ok, but if you want to get 0xn4 from n, then the register index should be 4n + 1, shouldn't it?

**Review by agkaminski — APPROVED** [review]
LGTM

**Pull request merged**